### DISCUSSION TITLE
Feature Request: Ability to modify the sampler, CFG scale and some other settings for ComfyUI image generator

### DISCUSSION THREAD

**Discussion by AlgorithmicKing737** [open-webui/open-webui#3491] [Ideas]
First of all i would like to say i am impressed by OpenWebUI's progress.

**Is your feature request related to a problem? Please describe.**
I can't adjust the sampler, CFG scale, and other settings in the ComfyUI image generator settings, that makes my image look extremely bad.

**Describe the solution you'd like**
Allow users to modify the sampler, CFG scale, schedular and seed.

**Describe alternatives you've considered**
ComfyUI's KSampler, include its every setting if possible.

**Additional context**
By ComfyUI image generator settings I mean this:
![Screenshot 2024-06-28 115727](https://github.com/open-webui/open-webui/assets/147901320/739bfa8e-e964-4977-9bb9-7e5372cbcbc2)
ComfyUI KSampler:
![Screenshot 2024-06-28 115818](https://github.com/open-webui/open-webui/assets/147901320/9a9ad7fe-dddb-4e05-9d7c-e09331c8965e)

**Comment by Geeknasty** [comment]
Yeah, I would also like to see this feature. It is possible for now to change these settings via environment variables as added in #3219 more details in the [docs](https://docs.openwebui.com/getting-started/env-configuration#image-generation)

However, it would be much more convenient to be able to change these settings via the settings.

**Comment by innovortex** [comment]
Hi,

The COMFYUI_CFG_SCALE variable does not appear to be used by **Comfyui + FLUX**. 
The generated images are with the default value of 3.5. 
By modifying the comfyui.py file in this way, I was able to achieve the desired result:

`
 "17": {
        "inputs": {
            "scheduler": "simple",
            "steps": 20,
            "cfg": 1,
            "denoise": 1,
            "model": [
                "12",
                0
            ]
        },
        "class_type": "BasicScheduler"
`
Do I miss something ?
Thx
